feat: implement homepage design

ShantelPeters opened the pull request:
# 🚀 Pull Request

## Description
This PR implements the homepage design with wallet connect component

## Related Issue
<!-- Add reference to the issue this PR addresses -->
Closes #16 

## Type of Change
<!-- Mark the appropriate option with an "x" -->
- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [x] 🔧 Refactoring
- [ ] 📚 Documentation update
- [x] 🧪 Test update
- [ ] 🔒 Security fix
- [ ] 🧹 Chore

## Proof of Completion
<!-- Provide evidence that the issue has been addressed properly -->
<!-- This could include screenshots, logs, or other relevant documentation -->
- Homepage Design:
![Screenshot from 2025-03-27 23-43-34](https://github.com/user-attachments/assets/1932fc9a-4fa2-4b29-96f5-ae52e23a6a21)

- Wallet Connect:
![Screenshot from 2025-03-30 17-16-17](https://github.com/user-attachments/assets/87087f7d-603b-4189-a0ad-5af4be8f5658)
![Screenshot from 2025-03-30 17-16-38](https://github.com/user-attachments/assets/055e707f-0b84-46eb-b2c1-4c79ce7746bd)
![Screenshot from 2025-03-30 17-16-47](https://github.com/user-attachments/assets/5d7df58e-4d7c-4377-b940-3a02c7466afa)
![Screenshot from 2025-03-30 17-17-32](https://github.com/user-attachments/assets/94b8c555-ba58-4cfa-9e22-ee777a423069)
![Screenshot from 2025-03-30 17-17-45](https://github.com/user-attachments/assets/ccf6fbe1-7f41-4001-b71f-875ec3c3ffc4)
![Screenshot from 2025-03-30 17-18-32](https://github.com/user-attachments/assets/fb220a61-8694-4a61-92a3-98d39e12ae3f)
![Screenshot from 2025-03-30 17-18-41](https://github.com/user-attachments/assets/ef286fbe-17aa-4aa3-aa1f-592cc9074266)


## Checklist
<!-- Mark items with an "x" when completed -->
- [x] ✅ My code follows the project's coding standards
- [ ] 📝 I have updated the documentation as needed
- [ ] 🧪 I have added tests that prove my fix/feature works
- [x] 🔄 I have rebased my branch with the latest main/develop
- [x] 👀 I have performed a self-review of my own code

## Additional Notes
<!-- Add any other information about the PR here -->

zleypner commented:
Hello @ShantelPeters ,
Please add a description to the PR, also link the PR with the issue to be closed. Thank you!
You can use the PR template:
https://github.com/TRADOXUS/tradoxus-frontend/blob/main/.github/pull-request-template.md

ShantelPeters commented:
Done @zleypner

zleypner reviewed (changes requested):
On regards of the issue #16 

It should be able to connect to the following

Stellar Wallet Integration
Coinbase Wallet Integration
MetaMask Integration

![image](https://github.com/user-attachments/assets/f33faed1-adde-4c3d-bc8f-ffdde023ee26)

zleypner commented:
@ShantelPeters Please let me know once is ready yo review. I will be pending.

ShantelPeters commented:
@zleypner I added wallet connection to the following wallets:

- Stellar Wallet (Rabet Wallet)
- Coinbase Wallet
- MetaMask

zleypner reviewed (changes requested):
Thank you

Currently it does not have connection with the wallet. Does not pop up the wallet asking for pw.  Even If the wallet is open it does not access to the wallet. it gives Error.  

Once it has been updated provide a Screenshot, as a probe it has connection to the wallets. Then I will proceed to test.

ShantelPeters commented:
@zleypner I have updated the `disconnect` function for connected wallets.
- Handled Rabet Wallet disconnection
- Handled Metasmask Wallet disconnection

For Coinbase Wallet, the `wallet_revokePermissions` method does not exist. So the disconnection will only disconnect the dApp on the frontend.

I have updated the PR with screenshots.

zleypner reviewed (changes requested):
Thank you! Please divide the wallet implementation into separate files for better maintainability per Layer. After this one it would be ready.

ShantelPeters commented:
Hello @zleypner , I have reorganized the wallet implementation into separate files for better maintainability as you have requested.

1. Types Layer (types/wallet.ts)
Contains all type definitions for the wallet system, including:
- Basic wallet types (WalletType, WalletState)
- UI-related types (WalletOption, WalletConnectionModalProps)
- Global window type extensions for wallet providers

2. API/Lib Layer
Split into two files for separation of concerns:
- `lib/wallet-verification.ts`: Handles verification of existing wallet connections
- `lib/wallet-connection.ts`: Manages the connection logic for different wallet types

3. Hook Layer (hooks/useWallet.ts)
Provides the main wallet hook that:
- Manages wallet state
- Loads saved wallet connections
- Handles connection/disconnection
- Sets up event listeners

ShantelPeters commented:
Hi ser @zleypner please kindly review requested changes..

zleypner commented:
Hello @ShantelPeters ,
The issue #29 , mainly requested  to support for Freighter, Albedo, and other popular Stellar wallets. Currently you installed rabet.

MPSxDev reviewed (changes requested):
Please solve errors: 
![image](https://github.com/user-attachments/assets/33d3e570-1db4-42a7-baf2-c285ef441a9f)

ShantelPeters commented:
@MPSxDev just a note, those errors are not from what I did. For the fonts specifically, here is where they were imported https://github.com/TRADOXUS/tradoxus-frontend/commit/1273cb64deba26c13d886924d1bcbcbed9125b23#diff-eca96d2c09f31517696a26e1d0be4070e1fbab02831481bed006e275741d030b

zleypner commented:
> Hello @ShantelPeters , The issue #29 , mainly requested to support for Freighter, Albedo, and other popular Stellar wallets. Currently you installed rabet.

@ShantelPeters We would pending on this mentioned previosly and resolving the conflicts

zleypner commented:
Hello @ShantelPeters , Could you please resolve conflicts?

ShantelPeters commented:
> Hello @ShantelPeters , Could you please resolve conflicts?
> 
> Other otherwise, I will proceed to close the PR. 
> 
> Best Regards

Okay sir .. on it

zleypner reviewed (changes requested):
Hey @ShantelPeters , I noticed that package-lock.json is locked to @next/swc-linux-x64-gnu, which causes issues on Windows. Could you please regenerate the lockfile to ensure cross-platform compatibility? Thanks!

ShantelPeters commented:
Hello @zleypner , I have regenerated the package lock file

zleypner commented:
I´ve run the PR and it seems have some errors that needs to be fixed. I be providing a better feedback tomorrow. Thank you
